### ISSUE TITLE
Couldn't read file LocalFile: '[...]/OpenCameraPlugin.kt' even though it exists. Please verify that this file has read permission and try again.

### ISSUE THREAD

**Issue by ghost** [openponce/opencamera#6]
Hello, when I run `flutter pub get` with `open_camera: ^0.0.5` I get this:
````
Couldn't read file LocalFile: 'D:\SDK\flutter\.pub-cache\hosted\pub.dartlang.org\open_camera-0.0.5\android\src\main\kotlin\org\openponce\open_camera\OpenCameraPlugin.kt' even though it exists. Please verify that this file has read permission and try again.
````

Is this a problem you can fix, or is it dependent of `flutter_tools`?

Refs:
[Couldn't read file LocalFile: *.kt' even though it exists. Please verify that this file has read permission and try again.](https://github.com/flutter/flutter/issues/47933)
[`flutter pub get` tries to read non-existing kotlin implementation](https://github.com/flutter/flutter/issues/42923)

**Comment by ghost** [comment]
Yeah, reading through the references it seems that it's a problem with `platform`...

**Comment by ghost** [comment]
Correct me if I'm wrong, but shouldn't the native implementation, i.e. android .java files, be in  java folder?
I've sniffed around the code, and found out that there's only java implmentation, but the package depends on kotlin as it is explicitly beign versioned and implemented in build.gradle files.

Maybe it's a problem with both the `platform` for not providing useful messages about what files should be expected in a kotlin implementation of the plugin and the plugin itself implementing kotlin in its build files thus causing `dart-sdk` to think it's a Kotlin implementation.

Furthermore, as an example, this was fixed in `cloud_funcions 0.4.1+2` with this pull: [Cloud function package name](https://github.com/FirebaseExtended/flutterfire/pull/1250/files)

**Comment by ghost** [comment]
Changing `main.java.srcDirs += 'src/main/kotlin'` → `main.java.srcDirs += 'src/main/java'` and renaming `src/main/kotlin` → `src/main/java` solves this.

It's because of this implementation:

https://github.com/flutter/flutter/blob/cb614d13394125ea8dcdc1ebcb7f03da22f59fc2/packages/flutter_tools/lib/src/platform_plugins.dart#L87-L106

**Comment by adminant** [comment]
Please help, I have the same problem when I trying to convert my Flutter app to plugin, adding plugin: and androidPackage and pluginClass fields to pub spec.yaml. After that it shows error like in theme topic though I don't have any kt files, and my java implementation is in app subdirectories of android. How can I get rid of this error?

**Comment by Jai05-techie** [comment]
hi @RickStanley where i need to do this changes  main.java.srcDirs += 'src/main/kotlin' → main.java.srcDirs += 'src/main/java' not getting ?

**Comment by ghost** [comment]
@Jai05-techie I don't remember exactly, but if you use any IDE/Text Editor that provides Search & Replace, do so.

I haven't touched this for a long time, I think that my previous workaround solved half the problem.
But I do remember starting a fresh plugin template and copying the algorithm from this project. I needed this plugin for a company that I worked. Unfortunately I don't have the repository nor the access anymore to share with you.

In other words: I think you are better off just starting a project anew and take this plugin's algorithm as a basis, or just copy the entire thing and see what doesn't work.

----
I'm going to leave this issue open for anyone who comes across this issue, but I'm going to unsubscribe as I'm not in touch with this.
